libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -4014,21 +4031,39 @@ string YulUtilFunctions::tryDecodeErrorMessageFunction()
 				if gt(length, 0xffffffffffffffff) { leave }
 
 				let end := add(add(msg, 0x20), length)
-				if gt(end, add(data, returndatasize())) { leave }
+				if gt(end, add(data, sub(returndatasize(), 4))) { leave }
```

Comment:
This is the bugfix. The bug allows malformed error data to be decoded as correct error data, which could lead to nondeterminism when this data is used.

Reply:
Malformed in the sense that the error data is too short for the length stored inside it.